pkg/client/subjectaccessreview.go

```diff
@@ -27,45 +23,49 @@ type SubjectAccessReviewInterface interface {
 }
 
-// subjectAccessReviews implements SubjectAccessReviewsNamespacer interface
+// subjectAccessReviews implements SubjectAccessReviews interface
 type subjectAccessReviews struct {
 	r     *Client
-	ns    string
 	token string
 }
 
-// newSubjectAccessReviews returns a subjectAccessReviews
-func newSubjectAccessReviews(c *Client, namespace, token string) *subjectAccessReviews {
+// newImpersonatingSubjectAccessReviews returns a subjectAccessReviews
+func newImpersonatingSubjectAccessReviews(c *Client, token string) *subjectAccessReviews {
 	return &subjectAccessReviews{
 		r:     c,
-		ns:    namespace,
 		token: token,
 	}
 }
 
-// Create creates new policy. Returns the server's representation of the policy and error if one occurs.
-func (c *subjectAccessReviews) Create(policy *authorizationapi.SubjectAccessReview) (result *authorizationapi.SubjectAccessReviewResponse, err error) {
-	result = &authorizationapi.SubjectAccessReviewResponse{}
-	err = overrideAuth(c.token, c.r.Post().Namespace(c.ns).Resource("subjectAccessReviews")).Body(policy).Do().Into(result)
-	return
-}
-
-// clusterSubjectAccessReviews implements ClusterSubjectAccessReviews interface
-type clusterSubjectAccessReviews struct {
-	r *Client
-}
-
-// newClusterSubjectAccessReviews returns a clusterSubjectAccessReviews
-func newClusterSubjectAccessReviews(c *Client) *clusterSubjectAccessReviews {
-	return &clusterSubjectAccessReviews{
+// newSubjectAccessReviews returns a subjectAccessReviews
+func newSubjectAccessReviews(c *Client) *subjectAccessReviews {
+	return &subjectAccessReviews{
 		r: c,
 	}
 }
 
-// Create creates new policy. Returns the server's representation of the policy and error if one occurs.
-func (c *clusterSubjectAccessReviews) Create(policy *authorizationapi.SubjectAccessReview) (result *authorizationapi.SubjectAccessReviewResponse, err error) {
+func (c *subjectAccessReviews) Create(sar *authorizationapi.SubjectAccessReview) (result *authorizationapi.SubjectAccessReviewResponse, err error) {
 	result = &authorizationapi.SubjectAccessReviewResponse{}
-	err = c.r.Post().Resource("subjectAccessReviews").Body(policy).Do().Into(result)
+
+	// if this a cluster SAR, then no special handling
+	if len(sar.Action.Namespace) == 0 {
+		err = overrideAuth(c.token, c.r.Post().Resource("subjectAccessReviews")).Body(sar).Do().Into(result)
+		return
+	}
+
+	err = c.r.Post().Resource("subjectAccessReviews").Body(sar).Do().Into(result)
+
+	// if the namespace values don't match then we definitely hit an old server.  If we got a forbidden, then we might have hit an old server
+	// and should try the old endpoint
+	if (sar.Action.Namespace != result.Namespace) || kapierrors.IsForbidden(err) {
+		deprecatedResponse := &authorizationapi.SubjectAccessReviewResponse{}
+		deprecatedAttemptErr := overrideAuth(c.token, c.r.Post().Namespace(sar.Action.Namespace).Resource("subjectAccessReviews")).Body(sar).Do().Into(deprecatedResponse)
+		if deprecatedAttemptErr == nil {
```

Comment:
ditto on non-nil deprecatedAttemptErr in the `(sar.Action.Namespace != result.Namespace)` case